graphql-cli/src/main/java/io/ballerina/graphql/generators/ICodeGenerator.java

```diff
@@ -0,0 +1,32 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package io.ballerina.graphql.generators;
+
+import io.ballerina.graphql.exceptions.BallerinaGraphqlException;
+import io.ballerina.graphql.exceptions.BallerinaGraphqlQueryValidationException;
+import io.ballerina.graphql.exceptions.BallerinaGraphqlSDLValidationException;
+
+/**
+ * CodeGenerator.
+ */
+public interface ICodeGenerator {
+    void generate(String sdlPath, String queriesPath, String outputPath)
```

Comment:
Let's add proper javadoc here. Whoever implements the interface should know what are the inputs and outputs and any constraints. 
https://stackoverflow.com/questions/9234571/javadoc-interface-comments

Reply:
This is resolved now & the changes are amended. Thank you.